content/en/blog/_posts/clientcmd-tutorial.md

```diff
@@ -0,0 +1,292 @@
+config, err := kubeConfig.ClientConfig()
+if err != nil {
+	// Do something
+}
+client, err := metav1.New(config)
+// ...
+```
+
+In the context of this article, there are six steps:
+
+1. [Configure the loading rules](#configure-the-loading-rules).
+1. [Configure the overrides](#configure-the-overrides).
+1. [Build a set of flags](#build-a-set-of-flags).
+1. [Bind the flags](#bind-the-flags).
+1. [Build the merged configuration](#build-the-merged-configuration).
+1. [Obtain an API client](#obtain-an-api-client).
+
+### Configure the loading rules
+
+`clientcmd.NewDefaultClientConfigLoadingRules()` builds loading rules which will use either the contents of the `KUBECONFIG` environment variable,
+or the default configuration file name (`~/.kube/config`).
+In addition, if the default configuration file is used,
+it is able to migrate settings from the (very) old default configuration file
+(`~/.kube/.kubeconfig`).
+
+You can build your own `ClientConfigLoadingRules`,
+but in most cases the defaults are fine.
+
+### Configure the overrides
+
+`clientcmd.ConfigOverrides` is a `struct` storing overrides which will be applied over the settings loaded from the configuration derived using the loading rules.
+In the context of this article,
+its primary purpose is to store values obtained from command line arguments.
+These are handled using the [pflag](https://github.com/spf13/pflag) library,
+which is a drop-in replacement for Go's [flag package](https://pkg.go.dev/flag),
+adding support for double-hyphen arguments with long names.
+
+In most cases there's nothing to set in the overrides;
+we will only bind them to flags.
+
+### Build a set of flags
+
+In this context, a flag is a representation of a command line argument,
+specifying its long name (such as `--namespace`),
+its short name if any (such as `-n`),
+its default value,
+and a description shown in the usage information.
+Flags are stored in instances of
+[the `FlagInfo` `struct`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#FlagInfo).
+
+Three sets of flags are available,
+representing the following command line arguments:
+
+* authentication arguments (certificates, tokens, impersonations, username/password);
+* cluster arguments (API server, certificate authority, TLS configuration, proxy, compression)
+* context arguments (cluster name, `kubeconfig` user name, namespace)
+
+The recommended selection includes all three with a named context selection argument and a timeout argument.
+
+These are all available using the `Recommended…Flags` functions.
+The functions take a prefix, which is prepended to all the argument long names.
+
+So calling
+[`clientcmd.RecommendedConfigOverrideFlags("")`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#RecommendedConfigOverrideFlags)
+results in flags including `--context`, `--namespace`, and so on.
+The `--timeout` flag is given a default of 0,
+and the `--namespace` flags has a corresponding short variant, `-n`.
```

Comment:
I recommend:
```suggestion
The `--timeout` argument is given a default of 0,
and the `--namespace` argument has a corresponding short variant, `-n`.
```

The actual things that you interact with as a user aren't flags, as least not as I see it.